transport-classes-io_uring/src/main/java/io/netty/channel/uring/AbstractIoUringStreamChannel.java

```diff
@@ -204,6 +205,17 @@ protected final void doRegister(ChannelPromise promise) {
         });
     }
 
+    @Override
+    protected Object filterOutboundMessage(Object msg) {
+        // Since we cannot use synchronous sendfile,
+        // the channel can only support DefaultFileRegion instead of FileRegion.
+        if (msg instanceof DefaultFileRegion && IoUring.isIOUringSpliceSupported()) {
```

Comment:
prepend `IORING_SPLICE_SUPPORTED` since there's no point to perform any filtering on the type if is not supported.